docs/proposals/inclusion-of-notes.md

```diff
@@ -0,0 +1,92 @@
+
+- [Introduction](#introduction)
+- [New features](#new-features)
+  - [Including a note](#including-a-note)
+  - [Include a section of a note](#include-a-section-of-a-note)
+  - [Include an attribute of a file (note property or frontmatter)](#include-an-attribute-of-a-file-note-property-or-frontmatter)
+  - [Inclusion of backlinks overview](#inclusion-of-backlinks-overview)
+  - [Introducing a query language](#introducing-a-query-language)
+
+## Introduction
+
+Initial work and thought on including a note was ignited by issue [#652](https://github.com/foambubble/foam/issues/652). Requested by a user was a likewise functionality as offered in Obsidian. This was simply the ability to include a note.
+
+Whilst researching digital gardening for my own setup, I came across an in-depth overview by [Maggie Appleton](https://maggieappleton.com/roam-garden). Showing examples of her personal Roam Research I see valuable possibilites to connect more information, if we would add additional functionalities to the possibility of including a note. This proposal displays these possible functionalities and markup.
+
+## New features
+
+### Including a note
+
+The minimal functionality is the ability to fully include a note. Markup used in Obsidian for this is `![[wikilink]]`. For Foam I would suggest to follow this syntax. Benefits being:
+
+- Adds minimal amount of knowledge required as syntax is based on the syntax of creating a wikilink.
+- Makes the auto-complete work ouf-of-the-box, without any additional code and listeners required.
+
+**Important**. A risk exists that a loop of including the same notes arises. E.g. Note A includes note B which includes note A. This needs to be prevented by the implementation and made visible to the user.
+
+### Include a section of a note
+
+It could be interesting to only include a section of a note instead of the entire note. In order to do so thse user should be able to use the following syntax:
+
+`![[wikilink#section-b]]`
+
+As a result it will include the section title + section content until the next section *or* end of file.
+
+### Include an attribute of a file (note property or frontmatter)
+
+As a user I could be interested in collecting the value of any given proeprty for a note. For example, I might want to include the tags as defined in the frontmatter of note A. This should be possible via the syntax:
+
+`![[wikilink:<property>]]`
+
+The property value should be lookedup by foam defined properties, e.g. title, **or** any property defined in the frontmatter of a note.
+
+So, the example of including the tags of a note should be:
+
+`![[wikilink:tags]]`
+
+### Inclusion of backlinks overview
+
+It could be interesting to easily display the backlinks of a note in the preview, rather then using the custom window foam provides. If an user wants to easily see, or perhaps format, the references of that note this could be an option.
+
+A syntax could look like:
+
+`!{{note-backlinks}}`
```

Comment:
This is trickier because it requires access to the graph data to compute the value.

Is this the rationale for changing the syntax here?

Reply:
I wanted to prevent clashing with a possible wikilink called `[[note-backlinks]]`. But, agree, it's a bit weird that the syntax changes. We could also treat backlinks as a property of the note. As such, you could have the syntax:

`[[wikilink:backlinks]]` 

That might simplify the implementation as well.

Reply:
We have had a few issues from users requesting this feature. We currently use `{{ ... }}` syntax for [variable substitution](https://wiki.dendron.so/notes/ba97866b-889f-4ac6-86e7-bb2d97f6e376.html#frontmatter-variable-substitution) since its the convention also adopted by many templating languages (eg. 11ty, jekyll, etc)

Currently we're thinking about having `{{ backlinks }}` be a reserved frontmatter variable that links to all current backlinks of a note

Reply:
That could be in an interesting solution. I'll put this in a separate proposal, including various options like the ones already mentioned.